### DISCUSSION TITLE
Some new features and suggestions of webots in underwater applications

### DISCUSSION THREAD

**Discussion by AaaGss** [cyberbotics/webots#6419] [Ideas]
Hello, development team. I'm a researcher working on underwater applications. Recently I have been using webots software, and I recently encountered several problems as follows:
1. Underwater vehicles are often used, but I did not find the existing model in the software, so I designed a simple one by myself, but I still sincerely hope that the development team can integrate a ready-made underwater vehicle in webots. If you want, I can open source mine to everyone.
2. There seems to be no simulation model such as sonar and forward looking sonar in the software, but this is very common in underwater, just like using cameras on land. I hope the development team can integrate forward looking sonar and sonar sensors in the software, which will undoubtedly make the ecology of the software better and better.
3. Since 2020a, the software has removed most resources in order to save volume, but every time it is used, it may have to download resources online, which is very bad, especially when the network environment is not good. So I sincerely hope that all resources can be packaged together, although the download time is a little longer, but the use will have a better experience. If you can, it is recommended to do two versions, one version with resources, generally without resources, so as to provide users with a better choice.

**Comment by ShuffleWire** [comment]
1) That would be great, imho :)  

2) https://github.com/cyberbotics/webots/discussions/6486

3) Seconding, it would be nice. Please note however that the cache allows to not download each time, only the first time you need a ressource. And I don't experience that much issue with that because I more often stay on the same world most of the time.

If the cache is not working, ensure it size it not too small : (1024 work fine for me)
![image](https://github.com/cyberbotics/webots/assets/83014015/abc571fe-a9dc-4bfb-913d-484acf3f09b0)

**Comment by AaaGss** [comment]
I have answered the second question. In response to the third question, although you said you only need to download once, which is great, but when online download, I can not download once. Network causes have been ruled out.




946175224
***@***.***



&nbsp;





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "cyberbotics/webots"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年3月19日(星期二) 下午4:30
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [cyberbotics/webots] Some new features and suggestions of webots in underwater applications (Discussion #6419)





  
 
#6486
 
 
 
Seconding, it would be nice. Please note however that the cache allows to not download each time, only the first time you need a ressource. And I don't experience that much issue with that because I more often stay on the same world most of the time.
 
  
If the cache is not working, ensure it size it not too small : (1024 work fine for me)
 image.png (view on web)
